cli/command/system/testdata/docker-info-plugins.golden

```diff
@@ -0,0 +1,56 @@
+Containers: 0
+ Running: 0
+ Paused: 0
+ Stopped: 0
+Images: 0
+Server Version: 17.06.1-ce
+Storage Driver: aufs
+ Root Dir: /var/lib/docker/aufs
+ Backing Filesystem: extfs
+ Dirs: 0
+ Dirperm1 Supported: true
+Logging Driver: json-file
+Cgroup Driver: cgroupfs
+Plugins:
+ Volume: local
+ Network: bridge host macvlan null overlay
+ Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
+Swarm: inactive
+Runtimes: runc
+Default Runtime: runc
+Init Binary: docker-init
+containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
+runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
+init version: 949e6fa
+Security Options:
+ apparmor
+ seccomp
+  Profile: default
+Kernel Version: 4.4.0-87-generic
+Operating System: Ubuntu 16.04.3 LTS
+OSType: linux
+Architecture: x86_64
+CPUs: 2
+Total Memory: 1.953GiB
+Name: system-sample
+ID: EKHL:QDUU:QZ7U:MKGD:VDXK:S27Q:GIPU:24B7:R7VT:DGN6:QCSF:2UBX
+Docker Root Dir: /var/lib/docker
+Debug Mode (client): false
+Debug Mode (server): true
+ File Descriptors: 33
+ Goroutines: 135
+ System Time: 2017-08-24T17:44:34.077811894Z
+ EventsListeners: 0
+Registry: https://index.docker.io/v1/
+Labels:
+ provider=digitalocean
+Experimental: false
+Insecure Registries:
+ 127.0.0.0/8
+Live Restore Enabled: false
+CLI Plugins:
```

Comment:
Ok, thinking if we should (can) put this in the `docker info` output; `docker info` is not available if no connection to the daemon can be made.

If we _do_ want to integrate it, I think we should split it into `Server` and `Client` (similar to `docker version` showing two sections, and in case we cannot connect to the server, we should print the CLI section only.

Reply:
The suggestions to include in `docker info` came from @silvin-lubecki. With the errors being in `docker help` now we could perhaps get away without this, or else I could split as you suggest if you both agree with that approach.

Reply:
I'm good with splitting the `docker info` command output into two sections. But I think the split should be done in another PR and rebase this one (which is already huge) on top of it. I spotted at least one info which is client only.
```
Debug Mode (client): false
```
WDYT?

Reply:
> in another PR 

That you are expecting me to do, right? (I'm ok with that, just want to know...)

> rebase this one (which is already huge)

There is a bunch of fairly generic preparatory cleanup which I've mostly floated to the front of the branch and could also easily (probably) pull out into another PR.

Reply:
Alternatively, we can drop the "Integrate CLI plugins into `docker info`" commit from this PR, and move it to a new PR that makes those changes, but if that complicates things significantly, I'm ok with merging the current state, and move it in a follow up (Ideally. we should make sure to have it done before a release is cut though, otherwise it could potentially break people)

Reply:
Thinking about this:

> docker info is not available if no connection to the daemon can be made.

So I think in case the connection cannot be made the behaviour of the "pretty print" mode is clear enough -- you print an error instead of the server info.

But what about `-f {{json .}}` mode? Should it output a JSON object where all the server related things are null? Where should the error be reported (should it be?).

I presume the JSON output is an API and so cannot be changed, e.g. to push the server info down underneath a `server` key.

These seem like complicated questions to answer (and this little threadlet doesn't seem like the right context)

So would it be ok for me to just statically adjust the pretty version here to separate server and client side info but defer the changes to handle printing client side info when the server is not available to a followup?

Reply:
See https://github.com/docker/cli/pull/1638 for an attempt to deal with this stuff.

Reply:
Now merged, I'm rebasing onto that now (don't expect it before Monday though)